File: Bernardo/exemplos/poly/java/Poly.java
//package polynomial;

public class Poly
{   
   public static void main(String [] args)
    {
        /* check commmand line */
        if (args.length != 3)
        {
            throw new RuntimeException("Wrong number of arguments");
        }

        /* extract coefficients from arguments */
        double [] coef = new double[3];        
        for (int i = 0; i < 3; i++)
        {
            try
            {
                coef[i] = Double.parseDouble(args[i]);
            }
            catch (Exception e)
            {
                throw new RuntimeException("Invalid argument, args[" + i + "]");
            }
        }

		/* compute roots */
        Polynomial poly = new Polynomial(coef);
        double roots[] = poly.getRoots();

        /* print results */
        System.out.print("Roots of (" + coef[0] + ", " + coef[1] + ", " + coef[2] + ") = \n");
        switch (roots.length)
        {
            case 0:
                System.out.print("  No roots\n");
                break;

            case 1:
                System.out.print ("  One root: " + roots[0] + "\n");
                break;

            case 2:
                System.out.print ("  Root 1: " + roots[0] + "\n");
                System.out.print ("  Root 2: " + roots[1] + "\n");
                break;
        }
    }
 
}

File: Bernardo/exemplos/poly/C/Poly.c
#include <stdio.h>
#include <stdlib.h>

#include "Polynomial.h"

int main(int argc, char* argv[])
{
    /* check commmand line */
    if (argc != 4)
    {
        fprintf(stderr, "Wrong number of arguments\n");
        exit(EXIT_FAILURE);
    }

    /* extract coefficients from arguments */
    double coef[3];
    int i;    
    for (i = 0; i < 3; i++)
    {
		if (sscanf(argv[i+1], "%lg", &coef[i]) !=1)
		{
			fprintf(stderr, "Invalid argument argv[%d] = \"%s\"\n", i, argv[i]);
			exit(EXIT_FAILURE);
		}
    }

	/* compute roots */
    struct roots r = getRoots(coef);

    /* print results */
    fprintf(stdout, "Roots of (%g, %g, %g) = \n", coef[0], coef[1], coef[2]);
    switch (r.length)
    {
        case 0:
            fprintf(stdout, "  No roots\n");
            break;

        case 1:
            fprintf (stdout, "  One root: %g\n", r.root[0]);
            break;

        case 2:
            fprintf (stdout, "  Root 1: %g\n", r.root[0]);
            fprintf (stdout, "  Root 2: %g\n", r.root[1]);
            break;
    }
	return 0;
}

